src/Type.h

```diff
@@ -734,18 +736,19 @@ class RecordType final : public Type {
     // Field initializations that can be deferred to first access,
     // beneficial for fields that are separately initialized prior
     // to first access.  Nil pointers mean "skip initializing the field".
-    std::vector<std::unique_ptr<detail::FieldInit>> deferred_inits;
+    std::vector<std::shared_ptr<detail::FieldInit>> deferred_inits;
 
     // Field initializations that need to be done upon record creation,
     // rather than deferred.  These are expressions whose value might
     // change if computed later.
     //
     // Such initializations are uncommon, so we represent them using
     // <fieldoffset, init> pairs.
-    std::vector<std::pair<int, std::unique_ptr<detail::FieldInit>>> creation_inits;
+    std::vector<std::pair<int, std::shared_ptr<detail::FieldInit>>> creation_inits;
```

Comment:
Seems fine to me. shared_ptr makes me shudder for other reasons, but I think it's not an issue as we only work with references of shared_ptrs elsewhere. An alternative could be a `Clone/Copy` on `FieldInit`, but not seeing the point.

Reply:
I'd like to understand your shuddering a bit better as I've absorbed this as a Perfectly Fine Idiom ... but I imagine that's better done separate from a PR review subthread 😏. I did consider making a copy of the `FieldInit` but yeah that seemed pretty heavy-handed.